### DISCUSSION TITLE
How to set body's background color?

### DISCUSSION THREAD

**Discussion by popmanhe** [ant-design/ant-design#40290] [Q&A] [answered]
I can use ConfigProvider to switch themes and colours. But ConfigProvider only controls the components within it. 

I'd like to change body's background colour using theme's token.

But tokens are not updated right away, when theme of ConfigProvider is modified. So I can't get the new tokens.

Is there a guild line that shows how to use token?

**Comment by arvinxx** [comment, marked as answer]
@popmanhe CP is only a data provider. So it don't have real dom to affect body. Maybe you can use [App](https://ant.design/components/app) Component to wrapper your app at top level.

```jsx
render(
<ConfigProvider theme={{ ... your config }}>
 <App>
    ...
 </App>
</ConfigProvider>
)
```


> Is there a guild line that shows how to use token?

we are preparing  styling guide and toolchain with it.

**Reply by hamudeshahin** [reply]
What about next.js ?

**Comment by popmanhe** [comment]
I should have made it more clear.
I was using following code:

```
const { token } = useToken();
useEffect(()=>{
     document.body.style.backgroundColor= token.colorBgContainer；
},[theme])
```


The issue I am having is that after I change the theme data of <ConfigProvider theme={{ ... your config }}>

The token is not changed, yet. So I still get the previous theme's background color. 

My question is how to get the changed token? So I can set the correct background color.

**Reply by arvinxx** [reply]
what about

```js
useEffect(()=>{
     document.body.style.backgroundColor= token.colorBgContainer；
},[theme, token])
```

**Comment by popmanhe** [comment]
```
useEffect(()=>{
     document.body.style.backgroundColor= token.colorBgContainer；
},[token])
```

This is actually working. Thanks.
